MapboxDirections/MBIntersection.swift

```diff
@@ -1,54 +1,55 @@
 import Foundation
 
+/**
+ A single cross street along a step.
+ */
 @objc(MBIntersection)
 public class Intersection: NSObject, NSSecureCoding {
     
     /**
      Index into bearings/entry array.
      
-     Used to calculate the bearing just before the turn. Namely, the clockwise angle from true north to the direction of travel immediately before the maneuver/passing the intersection. Bearings are given relative to the intersection. To get the bearing in the direction of driving, the bearing has to be rotated by a value of 180. The value is not supplied for depart maneuvers.
+     The index of the item in the headings array that corresponds to the road that the containing route step uses to approach the intersection.
     */
-    public let approachIndex: Int?
+    public var approachIndex: Int = -1
```

Comment:
We probably don’t need to allow clients to modify these values. So you can either change this back to `public let` and set it to `-1` in the initializer (as you seem to do anyways), or you can say `public private(set) var`.